STL_Extension/include/CGAL/assertions.h

```diff
@@ -89,18 +110,18 @@ inline bool possibly(Uncertain<bool> c);
 #else // no CGAL_NO_ASSERTIONS
 #  define CGAL_ASSERTIONS_ENABLED true
 #  define CGAL_assertion(EX) \
-   (CGAL::possibly(EX)?(static_cast<void>(0)): ::CGAL::assertion_fail( # EX , __FILE__, __LINE__))
+   ((!CGAL::get_use_assertions() || CGAL::possibly(EX))?(static_cast<void>(0)): ::CGAL::assertion_fail( # EX , __FILE__, __LINE__))
```

Comment:
We had that only for Polygon_2.  This was there as the Straight Skeleton code does orientation tests with non-simple polygons.  @MaelRL convinced me that this runtime test has negligible overhead as it is used only in debug mode.   On the other hand you are maybe right, because we have users who activate the assertions even in Release mode.

Reply:
I guess all it needs is to see how it is in practice, say for CDT2, Regularized Boolean operations, Mesh_3,    with and without the atomic bool in Debug and Release (with CGAL assertions enabled).

Reply:
what about adding a third mode? NO_ASSERTION, ASSERTIONS, RUNTIME_CONTROLLABLE_ASSERTIONS?